src/node.cc

```diff
@@ -4278,30 +4363,36 @@ void Init(int* argc,
     SafeGetenv("OPENSSL_CONF", &openssl_config);
 #endif
 
-  // Parse a few arguments which are specific to Node.
-  int v8_argc;
-  const char** v8_argv;
-  ParseArgs(argc, argv, exec_argc, exec_argv, &v8_argc, &v8_argv);
-
-  // TODO(bnoordhuis) Intercept --prof arguments and start the CPU profiler
-  // manually?  That would give us a little more control over its runtime
-  // behavior but it could also interfere with the user's intentions in ways
-  // we fail to anticipate.  Dillema.
-  for (int i = 1; i < v8_argc; ++i) {
-    if (strncmp(v8_argv[i], "--prof", sizeof("--prof") - 1) == 0) {
-      v8_is_profiling = true;
-      break;
+  std::string nodeopt;
+  if (SafeGetenv("NODEOPT", &nodeopt)) {
+    const char** argv_from_env = new const char*[(nodeopt.length()+1) / 2];
+    int argc_from_env = 0;
+    char* cstr = strdup(nodeopt.c_str());
```

Comment:
@sam-github `c_str()` is an alias for `data()` nowadays, it lives as long as the string instance itself